.github/workflows/test.yml

```diff
@@ -0,0 +1,185 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - ci/test/nightly
+  pull_request:
+
+jobs:
+
+  cache_info:
+    name: Bootstrap cache
+    runs-on: ubuntu-latest
+    outputs:
+      crates-io-index-head: ${{ steps.ls-crates-io-index.outputs.head }}
+      cargo-lock-hash: ${{ steps.hash-cargo-lock.outputs.hash }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - id: ls-crates-io-index
+        name: Get head commit hash of crates.io registry index
+        run: |
+          commit=$(
+            git ls-remote --heads https://github.com/rust-lang/crates.io-index.git master |
+            cut -f 1
+          )
+          echo "$commit"
+          echo "::set-output name=head::$commit"
+
+      - id: hash-cargo-lock
+        name: Calculate dependency cache key
+        run: |
+          hash=$(
+            ci/strip-own-version-from-cargo-lock.pl Cargo.lock |
+            sha1sum | cut -d ' ' -f 1
+          )
+          echo "$hash"
+          echo "::set-output name=hash::$hash"
+
+  update_deps:
+    name: Update dependencies
+    needs: cache_info
+    # Caches on Windows and Unix do not interop:
+    # https://github.com/actions/cache/issues/330#issuecomment-637701649
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      RUSTFLAGS: -D warnings
+      RUST_BACKTRACE: 1
```

Comment:
These environment variables look out of place here. I'd expect them to be in the `tests` job, however that one does not set them.

Reply:
Ouch, you're right, thanks for noticing that!